codice/ddf
# DDF-3664 Replacing Google Font in bootswatch themes with hosted fonts

#### What does this PR do?
Creates a ddf-theme module that overrides the external font url to replace that with hosted font css and font files. The ddf-theme module uses SASS to generate the CSS for the bootswatch themes using the SCSS files from bootswatch and bootstrap-sass, as well as CSS taken from the Google fonts, and the Google font files.
#### Who is reviewing it? 
@ahoffer @GabrielFabian 
#### Select relevant component teams: 
@codice/ui
#### Choose 2 committers to review/merge the PR. 
@bdeining
@jlcsmith 
#### How should this be tested?

To test this you should install DDF in experimental/development mode.

All of the following URL's should render correct css:
https://localhost:8993/admin/metrics
https://localhost:8993/admin/certificate
https://localhost:8993/admin
https://localhost:8993/admin/sources
https://localhost:8993/admin/geowebcache
https://localhost:8993/error
https://localhost:8993/admin/registry/local
https://localhost:8993/admin/registry/remote
https://localhost:8993/admin/queries
https://localhost:8993/admin/data-usage
https://localhost:8993/login
https://localhost:8993/idp
https://localhost:8993/logout
https://localhost:8993/admin/spatial-config
https://localhost:8993/search/standard

Additionally all of the following css files should be accessible:
https://localhost:8993/admin/certificate/css/ddf-theme-cosmo.css
https://localhost:8993/admin/certificate/css/ddf-theme-cyborg.css
https://localhost:8993/admin/certificate/css/ddf-theme-flatly.css
https://localhost:8993/admin/css/ddf-theme-cosmo.css
https://localhost:8993/admin/css/ddf-theme-cyborg.css
https://localhost:8993/admin/css/ddf-theme-flatly.css
https://localhost:8993/admin/data-usage/css/ddf-theme-cosmo.css
https://localhost:8993/admin/data-usage/css/ddf-theme-cyborg.css
https://localhost:8993/admin/data-usage/css/ddf-theme-flatly.css
https://localhost:8993/admin/geowebcache/css/ddf-theme-cosmo.css
https://localhost:8993/admin/geowebcache/css/ddf-theme-cyborg.css
https://localhost:8993/admin/geowebcache/css/ddf-theme-flatly.css
https://localhost:8993/admin/metrics/css/ddf-theme-cosmo.css
https://localhost:8993/admin/metrics/css/ddf-theme-cyborg.css
https://localhost:8993/admin/metrics/css/ddf-theme-flatly.css
https://localhost:8993/admin/queries/css/ddf-theme-cosmo.css
https://localhost:8993/admin/queries/css/ddf-theme-cyborg.css
https://localhost:8993/admin/queries/css/ddf-theme-flatly.css
https://localhost:8993/admin/registry/local/css/ddf-theme-cosmo.css
https://localhost:8993/admin/registry/local/css/ddf-theme-cyborg.css
https://localhost:8993/admin/registry/local/css/ddf-theme-flatly.css
https://localhost:8993/admin/registry/remote/css/ddf-theme-cosmo.css
https://localhost:8993/admin/registry/remote/css/ddf-theme-cyborg.css
https://localhost:8993/admin/registry/remote/css/ddf-theme-flatly.css
https://localhost:8993/admin/sources/css/ddf-theme-cosmo.css
https://localhost:8993/admin/sources/css/ddf-theme-cyborg.css
https://localhost:8993/admin/sources/css/ddf-theme-flatly.css
https://localhost:8993/admin/spatial-config/css/ddf-theme-cosmo.css
https://localhost:8993/admin/spatial-config/css/ddf-theme-cyborg.css
https://localhost:8993/admin/spatial-config/css/ddf-theme-flatly.css
https://localhost:8993/error/css/ddf-theme-cosmo.css
https://localhost:8993/error/css/ddf-theme-cyborg.css
https://localhost:8993/error/css/ddf-theme-flatly.css
https://localhost:8993/idp/css/ddf-theme-cosmo.css
https://localhost:8993/idp/css/ddf-theme-cyborg.css
https://localhost:8993/idp/css/ddf-theme-flatly.css
https://localhost:8993/login/css/ddf-theme-cosmo.css
https://localhost:8993/login/css/ddf-theme-cyborg.css
https://localhost:8993/login/css/ddf-theme-flatly.css
https://localhost:8993/logout/css/ddf-theme-cosmo.css
https://localhost:8993/logout/css/ddf-theme-cyborg.css
https://localhost:8993/logout/css/ddf-theme-flatly.css
https://localhost:8993/search/standard/css/ddf-theme-cosmo.css
https://localhost:8993/search/standard/css/ddf-theme-cyborg.css
https://localhost:8993/search/standard/css/ddf-theme-flatly.css

#### Any background context you want to provide?

Changes in the ddf-theme module should affect the CSS for every part of the app that was using bootswatch themes before. This means upgrading the bootswatch theme should also be easier.

#### What are the relevant tickets?
[](https://codice.atlassian.net/browse/)
#### Screenshots (if appropriate)
#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
